paderbox/utils/random_utils.py

```diff
@@ -25,73 +27,290 @@ def _force_correct_shape(f):
+
+    Args:
+        *shape:
+        loc:
+        scale:
+        truncation:
+        dtype:
 
-def normal(*shape, dtype=np.complex128):
-    return randn(*shape, dtype=dtype)
+    Returns:
+
+    >>> x = truncated_normal()
+    >>> x.ndim
+    0
+    >>> x = truncated_normal(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = truncated_normal(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return (
+        truncnorm(-truncation / scale, truncation / scale, loc, scale).rvs(shape)
+    )
 
 
 @_force_correct_shape
-def hermitian(*shape, data_type=np.complex128):
+@_add_kwarg_dtype(default=np.float64)
+def log_truncated_normal(*shape, loc=0., scale=.5, truncation=3.):
+    """
+
+    Args:
+        *shape:
+        loc:
+        scale:
+        truncation:
+        dtype:
+
+    Returns:
+
+    >>> x = log_truncated_normal()
+    >>> x.ndim
+    0
+    >>> x = log_truncated_normal(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = log_truncated_normal(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return np.exp(truncnorm(-truncation / scale, truncation / scale, loc, scale).rvs(shape))
+
+
+@_force_correct_shape
+@_add_kwarg_dtype(default=np.float64)
+def truncated_exponential(*shape, loc=0., scale=1., truncation=3.):
+    """
+
+    Args:
+        *shape:
+        loc:
+        scale:
+        truncation:
+
+    Returns:
+
+    >>> x = truncated_exponential()
+    >>> x.ndim
+    0
+    >>> x = truncated_exponential(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = truncated_exponential(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return truncexpon(truncation / scale, loc, scale).rvs(shape)
+
+
+@_force_correct_shape
+def hermitian(*shape, dtype=np.complex128):
     """ Assures a random positive-semidefinite hermitian matrix.
 
     :param shape:
-    :param data_type:
+    :param dtype:
     :return:
     """
-    assert shape[-1] == shape[-2]
-    matrix = uniform(shape, data_type)
+    assert len(shape) >= 2 and shape[-1] == shape[-2], shape
+    matrix = uniform(*shape, dtype=dtype)
     matrix = matrix + np.swapaxes(matrix, -1, -2).conj()
     np.testing.assert_allclose(matrix, np.swapaxes(matrix, -1, -2).conj())
     return matrix
 
 
 @_force_correct_shape
-def pos_def_hermitian(*shape, data_type=np.complex128):
+def pos_def_hermitian(*shape, dtype=np.complex128):
     """ Assures a random POSITIVE-DEFINITE hermitian matrix.
 
     TODO: Can this be changed? Why do we need 2?
 
     :param shape:
-    :param data_type:
+    :param dtype:
     :return:
     """
-    matrix = hermitian(*shape, data_type=data_type)
+    matrix = hermitian(*shape, dtype=dtype)
     matrix += np.broadcast_to(shape[-1] * 2 * np.eye(shape[-1]), shape)
     return matrix
+
+
+@dataclasses.dataclass
+class Uniform:
```

Comment:
I'd add a `dtype` attribute to all of these classes to match the function interfaces